content/200-concepts/100-components/01-prisma-schema/09-index-configuration.mdx

```diff
@@ -0,0 +1,143 @@
+
+- The `length` argument is available on MySQL on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` fields. It allows Prisma to now support indexes and constraints on `String` and `Bytes` types.
+
+- The `sort` argument is available for all databases on the `@unique`, `@@unique` and `@@index` fields. Additionally, SQL Server also allows it on `@id` and `@@id`.
+
+<Admonition type="warning">
+
+This might be a **breaking change** if you have a database that is already making use of these features, so it needs to be explicitly enabled via a Preview feature. See [Upgrading from previous versions](#upgrading-from-previous-versions) for details.
+
+</Admonition>
+
+In order to enable this as a Preview feature please add it to the `generator` block:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["extendedIndexes"]
+}
+```
+
+</TopBlock>
+
+## New `length` argument (MySQL)
+
+The `length` argument is specific to MySQL and will allow you to define indexes and constraints on columns of `String` and `Byte` types. For these types, MySQL requires you to specify a maximum length for the subpart of the value to be indexed in cases where the full value would exceed MySQL's limits for index sizes. See [the MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/innodb-limits.html) for more details.
+
+The `length` argument is available on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` fields.
+
+As an example, the following data model declares a field with a max length of 3000 characters as id.
+
+```prisma file=schema.prisma
+model Id {
+  id String @id @db.VarChar(3000)
+}
+```
+
+But this is not not valid in MySQL because it exceeds MySQL's index storage limit and therefore Prisma rejects the data model. The generated SQL would be rejected by the database.
+
+```sql
+Create Table `Id` (
+`id` varchar(3000) Primary Key
+)
+```
+
+With the new `length` property you can now specify that a subpart of the `id` value represents the primary key. In the example below the first 100 characters are used:
+
+```prisma file=schema.prisma
+model Id {
+  id String @id(length: 100) @db.VarChar(3000)
+}
+```
+
+Migrate is able to create constraints and indexes with these length arguments if specified in your data model. This means that you can now create indexes and constraints on values of Prisma type `Byte` and `String`. If you don't specify the argument the index is treated as covering the full value as before.
+
+Introspection will now fetch these limits where they are present in your existing database. This allows us to support indexes and constraints that were previously suppressed and will result in better support of existing MySQL databases that were making use of this feature.
+
+The `length` argument can also be used on compound primary keys, using the `@@id` field, as in the example below:
```

Comment:
This is this specifically spelled out here? As an example for compound things? Then maybe mention that this would also apply to @@unique and @@index this way?

Reply:
I've made a basic update to this by mentioning @@unique and @index below, it's still not great phrasing as I'm not sure what the correct terminology is to refer to all three together. Let me know if have an idea for something better but I'll resolve this now for the MVD version.